manifests/ppa.pp

```diff
@@ -20,52 +22,37 @@
   $sources_list_d_filename  = "${filename_without_ppa}-${release}.list"
 
   if $ensure == 'present' {
-    $package = $::lsbdistrelease ? {
-        /^[1-9]\..*|1[01]\..*|12.04$/ => 'python-software-properties',
-        default  => 'software-properties-common',
-    }
-
-    if ! defined(Package[$package]) {
-        package { $package: }
-    }
+    if $package_manage {
+      package { $package_name: }
 
-    if defined(Class[apt]) {
-        $proxy_host = $apt::proxy_host
-        $proxy_port = $apt::proxy_port
-        case $proxy_host {
-          false, '', undef: {
-            $proxy_env = []
-          }
-          default: {
-            $proxy_env = ["http_proxy=http://${proxy_host}:${proxy_port}", "https_proxy=http://${proxy_host}:${proxy_port}"]
-          }
-        }
+      $_require = [File['sources.list.d'], Package[$package_name]]
     } else {
-        $proxy_env = []
+      $_require = File['sources.list.d']
     }
-    exec { "add-apt-repository-${name}":
-        environment => $proxy_env,
-        command     => "/usr/bin/add-apt-repository ${options} ${name}",
-        unless      => "/usr/bin/test -s ${sources_list_d}/${sources_list_d_filename}",
-        user        => 'root',
-        logoutput   => 'on_failure',
-        notify      => Exec['apt_update'],
-        require     => [
-        File['sources.list.d'],
-        Package[$package],
-        ],
+
+    case $apt::proxy_host {
+      false, '', undef: {
+        $_proxy_env = []
+      }
+      default: {
+        $_proxy_env = ["http_proxy=http://${apt::proxy_host}:${apt::proxy_port}", "https_proxy=http://${apt::proxy_host}:${apt::proxy_port}"]
+      }
     }
 
-    file { "${sources_list_d}/${sources_list_d_filename}":
-        ensure  => file,
-        require => Exec["add-apt-repository-${name}"],
```

Comment:
That file is created by the magic that `add-apt-repository` does. Since we purge `sources.list.d` we need to declare a file resource for it otherwise it disappears.